template/cordova/lib/build.js

```
@@ -229,11 +268,66 @@ function buildTargets(msbuild, config) {
+            }
+            return msbuild.buildProject(path.join(ROOT, build.target), config.buildType,  build.arch, otherProperties);
          });
     }, Q());
+
+    if (shouldBundle) {
+        return buildsCompleted.then(function() {
+            // msbuild isn't capable of generating bundles unless you enable bundling for each individual arch
+            // However, that generates intermediate bundles, like "CordovaApp.Windows10_0.0.1.0_x64.appxbundle"
+            // We need to clear the intermediate bundles, or else "cordova run" will fail because of too
+            // many .appxbundle files.
+
+            console.log('Clearing intermediates...');
+            var appPackagesPath = path.join(ROOT, 'AppPackages');
+            var childDirectories = shell.ls(path.join(appPackagesPath, '*')).map(function(pathName) {
+                return { path: pathName, stats: fs.statSync(pathName) };
+            }).filter(function(fileInfo) {
+                return fileInfo.stats.isDirectory();
+            });
+
+            if (childDirectories.length === 0) {
+                throw new Error('Could not find a completed app package directory.');
+            }
+
+            // find the most-recently-modified directory
+            childDirectories.sort(function(a, b) { return b.stats.mtime - a.stats.mtime; });
+            var outputDirectory = childDirectories[0];
+
+            var finalFile = '';
+
+            var archSearchString = bundleTerms.replace(/\|/g, '_') + (config.buildType === 'debug' ? '_debug' : '') + '.appxbundle';
+            var filesToDelete = shell.ls(path.join(outputDirectory.path, '*.appx*')).filter(function(appxbundle) {
+                var isMatch = appxbundle.indexOf(archSearchString) === -1;
+                if (!isMatch) {
+                    finalFile = appxbundle;
+                }
+                return isMatch;
+            });
+            filesToDelete.forEach(function(file) {
+                shell.rm(file);
```

should we fail fast if we have trouble deleting a file?